### ISSUE TITLE
Question about the loss function

### ISSUE THREAD

**Issue by JaeDukSeo** [DuaneNielsen/DeepInfomaxPytorch#3]
![image](https://user-images.githubusercontent.com/22832406/52606300-21b7f780-2e40-11e9-889b-9dd1993779d3.png)

Is the fake and real term switched since we are maximizing the objective function?

**Comment by DuaneNielsen** [comment]
Hey Jae..

Let me think about that..

The equations used to make this term is eqn (4) in the paper.

![image](https://user-images.githubusercontent.com/11070508/52656943-55c90200-2eac-11e9-8cb1-ade002a555c0.png)

I think you have observed that we are taking Em -Ej, (Expectation of joint - Expectation of marginal), whereas the equation 4 defines I(X:E(X)) = Ej - Em.  I think you are right, we want to maximize I(X:E(X)) so we need to take the negative of it before we feed it to the loss function.

Maybe it would be clearer to write LOCAL = - (Ej - Em) * self.beta

Having said that, looking a bit deeper into things, there is a bit more work to really understand this.

As what we are optimizing is based on the referenced f-gan paper 

2.3 Variational Divergence Minimization (VDM)
We now use the variational lower bound (4) on the f-divergence Df (PkQ) in order to estimate a
generative model Q given a true distribution P.
To this end, we follow the generative-adversarial approach [10] and use two neural networks, Q and
T. Q is our generative model, taking as input a random vector and outputting a sample of interest.
We parametrize Q through a vector θ and write Qθ. T is our variational function, taking as input a
sample and returning a scalar. We parametrize T using a vector ω and write Tω.
We can learn a generative model Qθ by finding a saddle-point of the following f-GAN objective
function, where we minimize with respect to θ and maximize with respect to ω,

F(θ, ω) = Ex∼P [Tω(x)] − Ex∼Qθ[f∗(Tω(x))] . (6)

To optimize (6) on a given finite training data set, we approximate the expectations using minibatch
samples. To approximate Ex∼P [·] we sample B instances without replacement from the training set.
To approximate Ex∼Qθ
[·] we sample B instances from the current generative model Qθ.

https://arxiv.org/pdf/1606.00709.pdf

https://www.youtube.com/watch?v=y7pUN2t5LrA

So it's not actually Expectation of Joint - Expectation of marginal, rather its an adversarial loss based on minimizing the difference between a randomly selected image and the selected image, so that might be mis-named as well!

Have to run off.. will continue analysis later..

**Comment by JaeDukSeo** [comment]
One of the best answer I ever seen about this topic thanks!

**Comment by JaeDukSeo** [comment]
reading this again since it is so good
